### PR TITLE
Realtime segment count

### PR DESCRIPTION
**Pull request by franksmicro**
Update segmentCount in real time whenever a tag is added/modified (by calling updateTags).  Note that the previous pull request tag-segment-count needs to be implemented first, as this adds the new segmentCount field.  Also, the one-time update script in that pull request should be re-run once it is confirmed that segmentCount is being correctly updated in real time, in order to update the segementCount for tags added since the first time it was run.

### CODE REVIEW

File: scripts/tag-segment-count.js
Comment (JKHeadley) [inline scripts/tag-segment-count.js:1]: It looks like each PR is including code changes from a previous PR as well. Ideally each PR would be independent of each other to prevent merge conflicts, however being too strict about this can slow down development. In cases like this we just need to be careful to merge master into the PR branch to clear up any conflicts before merging the PR into master.

File: server/models/segment.model.js
Comment (JKHeadley) [inline server/models/segment.model.js]: Just want to reiterate the benefit of avoiding mixing format changes and past PR code changes with current code changes. In this case from what I can tell this code snippet is the only real change that needs reviewing in this PR, however it is difficult to tell due to all the formatting changes and code changes from previous/other PRs. Again, this is just something to keep in mind.

File: server/models/segment.model.js
Comment (JKHeadley) [inline server/models/segment.model.js:157]: It looks like there will be an issue here since only the tags from the "currentTags" list will be updated. This means any tags that were removed will only be in "oldTags" and therefore won't be updated.

I see two options to fix this:

1) Include the names for both `oldTags` and `currentTags`
2) Instead of updating them all in one loop, update each tag immediately after it is modified. This will also remove the need to query the db for the tags as you will already have them. Basically, for every tag in the `newTagsToAdd` or `existingTagsToAdd` lists, increase their segment count by 1, and do the opposite for the tags in the `deletedTags` list.

Comment (JKHeadley) [reply server/models/segment.model.js:157]: This is also a great example of how unit/e2e tests can be helpful. I've updated the tests to check for segment count changes when adding and removing tags. If you run them you will notice that they currently fail when tags are removed. You can see the changes I made to the test file in this PR.

You can run the tests by running the `run_e2e_tests.sh` script in the root project directory. If you need any help getting them going let me know.

File: test/e2e/end-to-end.test.js
Comment (JKHeadley) [inline test/e2e/end-to-end.test.js:342]: Note that this is the test that currently fails.

**Review by JKHeadley — CHANGES_REQUESTED** [review]
Most of the comments are just notes, but I did find an issue when removing tags. Details are in the code review comments. Let me know if you need any more info. Thanks so much!

File: server/models/segment.model.js
Comment (franksmicro) [reply server/models/segment.model.js:157]: Ok, I will pull down these tests.  I'll let you know if I have any issues running them.

I agree on the need for unit tests, but was not aware of what was out there so tested manually.  You are correct, I was thinking I only needed to update the tag count on new segments, thinking that the old segment was being deleted anyway.  But I see now that the old segment might still be referenced by other tags, and thus would not be deleted.  I'll work on changing the code to compute the counts for oldTags as well as currentTags.

File: test/e2e/end-to-end.test.js
Comment (franksmicro) [reply test/e2e/end-to-end.test.js:342]: Updated segment.model.js to update tag counts for both old (potentially deleted) and new segments.  All end to end tests are now passing.

**Review by JKHeadley — APPROVED** [review]
Looks good now! Feel free to merge when you're ready.

**Pull request merged**
